File: GSharp.PythonRegressionAnalysis/GSharp.Testbed/GlobalDataProvider.cs
using GHost.GSharp.Core;

namespace GSharp.Testbed
{
    public class GlobalDataProvider : GlobalDataProviderBase
    {
        public static GlobalDataProvider Instance = new GlobalDataProvider();

        public static void Initialize() => GlobalData.GlobalDataProvider = Instance;


        public override string ProductTitle => "gSharp Developer Testbed";

        public override string ProductVersionString => Versions.ProductVersion;

        public override string LogSource => "$DevTestbed";

        public override string ProductName => "GHost.GSharp.Developer.Testbed";

        public override string CompanyDataFolder => SolutionConstants.CompanyDataFolder;
    }
}

File: Samples/PISystem/VB.NET/GSharp.PythonRegressionAnalysis/GSharp.Testbed/GlobalDataProvider.vb
Imports GHost.GSharp.Core

Public Class GlobalDataProvider
    Inherits GlobalDataProviderBase

    Public Shared Instance As New GlobalDataProvider()

    Public Shared Sub Initialize()
        GlobalData.GlobalDataProvider = Instance
    End Sub

    Public Overrides ReadOnly Property ProductTitle As String
        Get
            Return "gSharp Developer Testbed"
        End Get
    End Property

    Public Overrides ReadOnly Property ProductVersionString As String
        Get
            Return Versions.ProductVersion
        End Get
    End Property

    Public Overrides ReadOnly Property LogSource As String
        Get
            Return "$DevTestbed"
        End Get
    End Property

    Public Overrides ReadOnly Property ProductName As String
        Get
            Return "GHost.GSharp.Developer.Testbed"
        End Get
    End Property

    Public Overrides ReadOnly Property CompanyDataFolder As String
        Get
            Return SolutionConstants.CompanyDataFolder
        End Get
    End Property
End Class
